src/runtime/providers/imgproxy.ts

```diff
@@ -0,0 +1,77 @@
+import { joinURL } from "ufo";
+import { createOperationsGenerator } from "@nuxt/image/dist/runtime/utils/index";
+import type { ProviderGetImage } from "@nuxt/image";
+import { Buffer } from "buffer";
+import * as hash from "hash.js";
+
+const operationsGenerator = createOperationsGenerator({
+  keyMap: {
+    resize: "rs",
+    size: "s",
+    fit: "rt",
+    width: "w",
+    height: "h",
+    dpr: "dpr",
+    enlarge: "el",
+    extend: "ex",
+    gravity: "g",
+    crop: "c",
+    padding: "pd",
+    trim: "t",
+    rotate: "rot",
+    quality: "q",
+    maxBytes: "mb",
+    background: "bg",
+    backgroundAlpha: "bga",
+    blur: "bl",
+    sharpen: "sh",
+    watermark: "wm",
+    preset: "pr",
+    cacheBuster: "cb",
+    stripMetadata: "sm",
+    stripColorProfile: "scp",
+    autoRotate: "ar",
+    filename: "fn",
+    format: "f",
+  },
+  formatter: (key, value) => `${key}:${value}`,
+});
+
+function urlSafeBase64(string: string) {
+  return Buffer.from(string, "utf8")
+    .toString("base64")
+    .replace(/=/g, "")
+    .replace(/\+/g, "-")
+    .replace(/\//g, "_");
+}
+
+const hexDecode = (hex: string) => Buffer.from(hex, "hex");
+
+function sign(salt: string, target: string, secret: string) {
+  const hmac = hash.hmac(hash.sha256, hexDecode(secret));
+  hmac.update(hexDecode(salt));
+  hmac.update(target);
+
+  return urlSafeBase64(hmac.digest());
+}
+
+const defaultModifiers = {
+  fit: "fill",
+  width: 0,
+  height: 0,
+  gravity: "no",
+  enlarge: 1,
+  format: "webp",
+};
+
+export const getImage: ProviderGetImage = (src, options) => {
+  const { modifiers, url, salt, key } = options;
+  const mergeModifiers = { ...defaultModifiers, ...modifiers };
```

Comment:
As per the [Cloudinary provider](https://github.com/nuxt/image/blob/3a98a685746b5be2188ddf59d091a4f3e525067b/src/runtime/providers/cloudinary.ts#L86), `defu` should be used here to ensure the defaults correctly merge.

`<NuxtImg src="..." width="400" height="400" />` results in the following `mergeModifiers` values with and without `defu`, if `modifiers` isn't set in nuxt.config.ts.

Without:
```
{ fit: undefined, width: 400, height: 400, gravity: 'no', enlarge: 1, format: undefined, quality: undefined, background: undefined }
{ fit: undefined, width: 800, height: 800, gravity: 'no', enlarge: 1, format: undefined, quality: undefined, background: undefined }
{ fit: 'fill', width: 400, height: 400, gravity: 'no', enlarge: 1, format: 'webp' }
```

With:
```
{ fit: 'fill', width: 400, height: 400, gravity: 'no', enlarge: 1, format: 'webp' }
{ fit: 'fill', width: 800, height: 800, gravity: 'no', enlarge: 1, format: 'webp' }
{ fit: 'fill', width: 400, height: 400, gravity: 'no', enlarge: 1, format: 'webp' }
```